sklearn/metrics/pairwise.py

```diff
@@ -1059,13 +1064,14 @@ def _parallel_pairwise(X, Y, func, n_jobs, **kwds):
     if effective_n_jobs(n_jobs) == 1:
         return func(X, Y, **kwds)
 
-    # TODO: in some cases, backend='threading' may be appropriate
-    fd = delayed(func)
-    ret = Parallel(n_jobs=n_jobs, verbose=0)(
-        fd(X, Y[s], **kwds)
+    # enforce a threading backend to prevent data communication overhead
+    fd = delayed(_dist_wrapper)
+    ret = np.empty((X.shape[0], Y.shape[0]), dtype=X.dtype)
```

Comment:
I would be curious to know if allocating this array Fortran ordered for better cache locality would matter for performance (no need to run all benchmarks, just one typical case).

Reply:
nice one, but actually, is there any reason why we slice on `Y` and not on `X`? Otherwise we can swap, and keep the C order.

Reply:
The thing is in `pairwise_distances_chunked` we are chunking along `X` and then `X_chunk` and `Y` are sent to `pairwise_distances` that can be parallelized with `n_jobs`. So we cannot chunk both `pairwise_distances` and  `pairwise_distances_chunked`  in the same way. 

Maybe this is something we can think about in a follow-up PR.